src/main/java/Parser.java

```diff
@@ -0,0 +1,206 @@
+        }
+
+        if (isToDo(fullCommand)) {
+            String taskType = "todo";
+            String taskDescription = processTaskDescription(fullCommand);
+
+            Command addCommand = new AddCommand(taskType, taskDescription);
+
+            return addCommand;
+        } else if (isDeadline(fullCommand)) {
+            String[] splitBySlash = fullCommand.split("/");
+
+            String taskType = "deadline";
+            String taskDateTime = processDateTime(splitBySlash);
+            String taskDescription = processTaskDescription(splitBySlash[0]);
+
+            Command addCommand = new AddCommand(taskType, taskDescription, taskDateTime);
+
+            return addCommand;
+        } else {
+            String[] splitBySlash = fullCommand.split("/");
+
+            String taskType = "event";
+            String taskDateTime = processDateTime(splitBySlash);
+            String taskDescription = processTaskDescription(splitBySlash[0]);
+
+            Command addCommand = new AddCommand(taskType, taskDescription, taskDateTime);
+
+            return addCommand;
+        }
+    }
+
+    /**
+     * Processes a find command for a word.
+     * @param fullCommand The user input for the find command.
+     * @return The find command.
+     */
+    private static Command processFindCommand(String fullCommand) {
+        String[] splitCommand = fullCommand.split(" ");
+
+        String word = splitCommand[1];
+
+        Command findCommand = new FindCommand(word);
+
+        return findCommand;
+    }
+
+    /**
+     * Splits the input String into appropriate format.
+     * @param input The user input.
+     * @return The description of the action to be carried out.
+     */
+    private static String processTaskDescription(String input) {
+        String[] splitInput = input.split(" ");
+        int arrSize = splitInput.length;
+
+        String returnString = splitInput[1];
+
+        for (int i = 2; i < arrSize; i++) {
+            returnString += " " + splitInput[i];
+        }
+
+        return returnString;
+    }
+
+    /**
+     * Processes the date time into appropriate format.
+     * @param splitBySlash The split input.
+     * @return The appropriate date time format.
+     */
+    private static String processDateTime(String[] splitBySlash) {
+        String day = splitBySlash[1];
+        String month = splitBySlash[2];
+
+        String[] splitYearTime = splitBySlash[3].split(" ");
+
+        String year = splitYearTime[0];
+        String time = splitYearTime[1];
+
+        DateTime dateTime = new DateTime(day, month, year, time);
+        dateTime.format();
+
+        return dateTime.toString();
+    }
+
+    private static boolean isBye(String str) {
+        return str.equals("bye") || str.equals("Bye");
+    }
+
+    private static boolean isList(String str) {
+        return str.equals("list");
+    }
+
+    private static boolean isDone(String str) {
+        return str.startsWith("done");
```

Comment:
I feel that it might be better to check if `str.split(" ")[0].equals("done")`, as this implementation my falsely accept commands like `doneextrawords`, which I don't feel it should.